src/jit.rs

```diff
@@ -1190,8 +1184,12 @@ impl<'a, C: ContextObject> JitCompiler<'a, C> {
         }
 
         if self.config.enable_address_translation {
-            let access_type = if value.is_none() { AccessType::Load } else { AccessType::Store };
-            let anchor = ANCHOR_TRANSLATE_MEMORY_ADDRESS + len.trailing_zeros() as usize + 4 * (access_type as usize);
+            let access_type = match value {
```

Comment:
Calling this `access_type` may be confusing with `AccessType`. What about altering the enum to represent load, store reg and store imm?

Reply:
`AccessType` comes from the memory mapping and that does not know about store immediate.
So I renamed `access_type` to `anchor_base`.